### PR TITLE
Sync: Filter out callbacks from post types and taxonomies callables

### PR DESCRIPTION
**Pull request by ebinnion**
We had a report of the following warnings and error:

```
[21-Aug-2016 15:53:14 UTC] PHP Fatal error: Allowed memory size of 134217728
bytes exhausted (tried to allocate 130968 bytes) in
/wp-content/plugins/jetpack/sync/class.jetpack-sync-json-deflate-codec.php on
line 39

[21-Aug-2016 15:58:25 UTC] PHP Warning: unserialize(): Function
spl_autoload_call() hasn't defined the class it was called for in
/wp-content/plugins/jetpack/sync/class.jetpack-sync-json-deflate-codec.php on
line 17

[21-Aug-2016 20:11:59 UTC] PHP Fatal error: Maximum execution time of 300
seconds exceeded in
/wp-content/plugins/jetpack/sync/class.jetpack-sync-json-deflate-codec.php on
line 38
```

The user did some debugging and was able to narrow the issue down to the fact that they have plugins that use the `meta_box_cb` argument when registering taxonomies. When the user turned off taxonomies, errors went away.

Further, the user reported that the infinite loops were likely due to circular references where the callback pointed back to the post type or taxonomy and around and around it went.

Since we are likely not interested in these callbacks, let's not sync them.

To test:

- Run tests locally

cc @lezama for review

### CODE REVIEW

File: sync/class.jetpack-sync-functions.php [hunk 1]
```diff
@@ -14,14 +14,40 @@ public static function get_modules() {
...
 	}
 
 	public static function get_post_types() {
 		global $wp_post_types;
+		$cloned_post_types = array();
+
+		foreach ( (array) $wp_post_types as $key => $post_type ) {
+			$cloned_post_type = new stdClass();
+			foreach ( (array) get_object_vars( $post_type ) as $var_key => $value ) {
+				if ( 'register_meta_box_cb' == $var_key ) {
```
Comment (lezama) [inline sync/class.jetpack-sync-functions.php]: let's use `===` here.

Comment (ebinnion) [inline sync/class.jetpack-sync-functions.php]: I'm not sure that there's much of a benefit for that here.

**Comment by kraftbj** [comment]
Tested this on a WP Engine site currently seeing the fatals matching this error and the fatals continued.

**Comment by lezama** [comment]
@kraftbj could you test again with master? this https://github.com/Automattic/jetpack/pull/5013 should fix it

File: sync/class.jetpack-sync-functions.php (hunk 1 above)
Comment (samhotchkiss) [inline sync/class.jetpack-sync-functions.php]: `===` is cheaper, right?  Since `==` tries to cast them to the same type, but `===` give up earlier

**Comment by lezama** [comment]
We should close this one, once @kraftbj confirms he doesn't get the errors when running master, in the meanwhile I am moving this to 4.3.1

**Pull request closed without merging**